### PR TITLE
Add test for overflow during multiplication in TestParseDurationSeconds.

### PR DESCRIPTION
**Pull request by sgmiller**
Error out rather than producing negative values when this happens.  Addresses https://hashicorp.atlassian.net/browse/VAULT-5437

### CODE REVIEW

**Review by jefferai — APPROVED** [review]

File: parseutil/parseutil.go [hunk 1]
```diff
@@ -114,7 +117,7 @@ func ParseDurationSecond(in interface{}) (time.Duration, error) {
 		}
 
 		if v, err := strconv.ParseInt(inp, 10, 64); err == nil {
-			return time.Duration(v) * time.Second, nil
+			return overflowMul(time.Duration(v), time.Second)
```
Comment (stevendpclark) [inline parseutil/parseutil.go:120]: Shouldn't the other multiplications below be modified to leverage the function?

**Review by stevendpclark — APPROVED** [review]
